python/ray/data/dataset.py

```diff
@@ -1171,6 +1171,63 @@ def split_proportionately(self, proportions: List[float]) -> List["Dataset[T]"]:
 
         return self.split_at_indices(split_indices)
 
+    def train_test_split(
+        self,
+        test_size: Union[int, float],
+        *,
+        shuffle: bool = False,
```

Comment:
Shuffle is a transformation while splitting is a consumption. Can this be dropped and let user do `ds.shuffle().train_test_split()`?